modules/consensus/consensusset.go

```diff
@@ -176,7 +177,10 @@ func (cs *ConsensusSet) ChildTarget(id types.BlockID) (target types.Target, exis
 func (cs *ConsensusSet) Close() error {
 	cs.mu.Lock()
 	defer cs.mu.Unlock()
-	return cs.db.Close()
+	return build.JoinErrors([]error{
+		build.AddContext("log.Close failed:", cs.log.Close()),
+		build.AddContext("db.Close failed:", cs.db.Close()),
+	}, "; ")
```

Comment:
Variadic appeals to me too, although `build.JoinErrors` currently has nice symmetry with `strings.Join`.

Ooh, something else occurred to me: if `build.JoinErrors` only sees one non-nil error, it should return that error directly instead of creating a new error. This preserves the ability to check for things like `io.EOF`.